core/indigo-core/molecule/molecule_cdxml_loader.h

```diff
@@ -0,0 +1,204 @@
+#include "elements.h"
+#include "molecule/base_molecule.h"
+#include "molecule/molecule_stereocenter_options.h"
+#include "molecule/query_molecule.h"
+
+typedef unsigned short int UINT16;
+typedef int INT32;
+typedef unsigned int UINT32;
+#include "molecule/CDXConstants.h"
+
+class TiXmlHandle;
+class TiXmlElement;
+class TiXmlNode;
+class TiXmlAttribute;
+
+namespace indigo
+{
+    class Scanner;
+    class Molecule;
+    class QueryMolecule;
+
+    class AutoInt
+    {
+    public:
+        AutoInt() : val(0){};
+
+        AutoInt(int v) : val(v){};
+
+        AutoInt(const std::string& v) : val(std::stoi(v))
+        {}
+
+        operator int() const
+        {
+            return val;
+        }
+
+        operator std::string() const
+        {
+            return std::to_string( val );
+        }
+
+    private:
+        int val;
+    };
+
+
+    struct _ExtConnection
+    {
+        int bond_id;
+        int point_id;
+        int atom_idx;
+    };
+
+    struct CdxmlNode
+    {
+        CdxmlNode() : element(ELEM_C), type( kCDXNodeType_Element ) // Carbon by default
+        {
+        }
+        AutoInt id;
+        std::string label;
+        AutoInt element;
+        Vec3f pos;
+        int type;
+        AutoInt isotope;
+        AutoInt charge;
+        AutoInt radical;
+        AutoInt valence;
+        AutoInt hydrogens;
+        AutoInt stereo;
+        AutoInt enchanced_stereo;
+        AutoInt stereo_group;
+        AutoInt index;
+        bool is_not_list;
+        std::vector<AutoInt> element_list;
+        std::unordered_map<int, int> bond_id_to_connection_idx;
+        std::unordered_map<int, int> node_id_to_connection_idx;
+        std::vector<_ExtConnection> connections;
+    };
+
+    struct CdxmlBond
+    {
+        CdxmlBond() : order(1)
+        {
+        }
+        AutoInt id;
+        std::pair<AutoInt, AutoInt> be;
+        AutoInt order;
+        AutoInt stereo;
+        AutoInt dir;
+        bool swap_bond;
+    };
+
+    struct CdxmlBracket
+    {
+        CdxmlBracket() : repeat_pattern(RepeatingUnit::HEAD_TO_TAIL)
+        {
+        }
+        std::vector<AutoInt> bracketed_list;
+        int usage;
+        AutoInt repeat_count;
+        int repeat_pattern;
+        std::string sru_label;
+    };
+
+
+    inline std::vector<std::string> split(const std::string& str, char delim)
+    {
+        std::vector<std::string> strings;
+        size_t start;
+        size_t end = 0;
+        while ((start = str.find_first_not_of(delim, end)) != std::string::npos)
+        {
+            end = str.find(delim, start);
+            strings.push_back(str.substr(start, end - start));
+        }
+        return strings;
+    }
+
+    class MoleculeCdxmlLoader
+    {
+    public:
+        DECL_ERROR;
+
+        MoleculeCdxmlLoader(Scanner& scanner);
+
+        void loadMolecule(BaseMolecule& mol);
+
+        StereocentersOptions stereochemistry_options;
+        bool ignore_bad_valence;
+        bool _has_bounding_box;
```

Comment:
is not initialized by default. (is not assigned in constructor and does not have own default constructor)